### ISSUE TITLE
Tasker/Automate/MacroDroid plugin

### ISSUE THREAD

**Issue by lublak** [florisboard/florisboard#332]
Its a bit off and i think very complex, but currently there is no keyboard available that has tasker support.

___

Events like (could also have a flag with: stop default handling, like if you Key pressed, Key Event triggered but not send a key to the text field):

- Key Event
- Autocomplete Event (if you click on a suggestion or on autocomplete, if implemented)
- Opened
- Closed
- Copied/Pasted/Cutted to/from Clipboard

Action:
- Send key
- Send text
- Text selection range
- Open/Close keyboard
- Copy/Paste/Cut Clipboard (in feature with the index number?)

- Change settings? Like language

State:
- get current text
- get current clipboard text(in the feature as an array?)
- get current suggestions

- get settings?

**Comment by patrickgold** [comment]
Thanks for your suggestion!

This is definitely something more complex than other to be implemented features and I am currently unsure if I will add this to FlorisBoard or not. If I decide to do so, it will definitely be not before v0.5.0 is released. I am adding this to my maybe todo list.

**Comment by lublak** [comment]
Yes I know. I think it could be useful, but also it is very complex to implement. Also I don't know how many people would use this feature.

Some tasker/automate (I use automate) examples would be:
- Text expand with dynamic content like date or some device Infos
- Switch language in different apps, like programming I use mostly in English but privat I use German
- auto suggest some dynamic contents, like text expand
- sync clipboard with desktop device 
- change to one hand mode if you are in walk

**Comment by lawson85** [comment]
I also agree that Tasker/MacroDroid should be supported. I have the app at the moment and cannot see it in the plugin list. Maybe I am missing something

**Comment by lublak** [comment]
@lawson85 No, you are not missing anything.
It's e feature request. So it's not implemented currently.
Just a: It would be cool if it supports tasker.

**Comment by lawson85** [comment]
It looks like this code can be more or less copy/pasted according to Tasker developer:
https://tasker.joaoapps.com/pluginslibrary.html

**Comment by lublak** [comment]
@lawson85 
it is already more complex. The interface itself, perhaps. But each individual function must first be passed on to the interface.
That's a lot of work.
